ipaserver/plugins/certmap.py

```diff
@@ -91,6 +91,49 @@
 register = Registry()
 
 
+def check_maprule_is_for_trusted_domain(api_inst, options):
+    """
+    Check that the certmap rule references altSecurityIdentities and
+    associateddomain in options is a trusted domain.
+
+    :param api_inst: API instance
+    :param options: options passed to the command
+                    at least ipacertmapmaprule and
+                    associateddomain options are expected for the check
+
+    :raises: ValidationError if altSecurityIdentities is present in the
+             rule but no Active Directory trusted domain listed in
+             associated domains
+    """
+    is_trusted_domain_required = False
+    maprules = options.get('ipacertmapmaprule')
+    if maprules:
+        for maprule in maprules:
+            if 'altSecurityIdentities' in maprule:
```

Comment:
Just curios, should this be case-insensitive?

Reply:
yes, the search should be case-insensitive. Good catch.